src/lib/yaraparse.py

```diff
@@ -153,20 +159,114 @@ def ursify_nocase_bytes(raw: bytes) -> UrsaExpression:
     return UrsaExpression(f"{{{ ' '.join(out) }}}")
 
 
-def ursify_plain_string(string: PlainString) -> UrsaExpression:
-    text_ascii = string.pure_text
-    text_wide = bytes(x for y in text_ascii for x in [y, 0])
+def encode_wide_bytes(raw: bytes) -> bytes:
+    return bytes(x for y in raw for x in [y, 0])
+
+
+def flatten_regex_or_tree(unit: Any) -> Optional[List[bytes]]:
+    if type(unit) is RegexpText:
+        return [unit.text.encode()]
+    elif type(unit) is RegexpOr:
+        left = flatten_regex_or_tree(unit.left)
+        right = flatten_regex_or_tree(unit.right)
+        if not left or not right:
+            return None
+        return left + right
+    elif type(unit) is RegexpConcat:
+        chars = [flatten_regex_or_tree(u) for u in unit.units]
+        string = b""
+        for c in chars:
+            if c is None:
+                return None
```

Comment:
shouldn't it be `continue`, and shouldn't we return more than one string here? Think, something like:

```
asdf(^&*)qwer
```
(where `^&*` represents anything we don't currently support)

the result should be `"asdf" & "qwer"` instead of None IMO

More practical example:
```python
strings: $a = /(tests? code)/
```